Automattic/vip-go-mu-plugins
# Fix image sizes and srcset for intermediate images

## Description
This PR uses a new filter available in WP 5.7 to properly pass image dimensions for the image `srcset` and `sizes`.
This needs to be tested and merged in coordination with the WP 5.7 release (March 9 is the current tentative release date).

New filter added in WP 5.7:  https://core.trac.wordpress.org/browser/trunk/src/wp-includes/media.php?rev=50136#L1643

How the fix works:
- First we hook into `wp_img_tag_add_width_and_height_attr` which is early enough that our image source still has the query arguments available so we know what the image size is intended to be
- We save that image size data and then pass it into `wp_image_src_get_dimensions` where we set image dimensions
- If the image is a full size image, we set the default dimensions
- If the image is not full size, we calculate the image size based on the given width and the calculated aspect ratio.

## Changelog Description

#### Bug Fix: Fixed an issue where images were not properly sized when using the Block Editor and intermediate image sizes.

Previously, all images added through the Block Editor that were assigned intermediate image sizes ( e.g. Thumbnail ) would appear as full size images.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1. Patch your `wp-includes/media.php` file to include the changes to `wp_image_src_get_dimensions` found here: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/media.php?rev=50136#L1601
2. Apply this code to your sandbox
3. Make sure your test site has Gutenberg Enabled
4. Create a new Post
5. Select the Image Block, add an image
6. From the Block Properties, select an intermediate image size like Thumbnail
7. Add more image blocks for Full Size, Medium, Large, etc..
8. Verify that the `srcset` and `sizes` attributes output are correct.

Before the change:
![before-patch-srcset](https://user-images.githubusercontent.com/3220162/109569248-fa6f0a00-7a9c-11eb-9788-fc2e3f700387.jpg)

After the change:
![after-patch-srcset](https://user-images.githubusercontent.com/3220162/109569279-09ee5300-7a9d-11eb-8824-2e3461a3d2bf.jpg)
